aip/general/0180.md

```diff
@@ -139,6 +139,30 @@ code, as such changes will be seen as breaking by those users.
 whether a proposed change is likely to break users, and an expansive reading of
 this guidance could ostensibly prevent _any_ change (which is not the intent).
 
+#### Field behavior compatibility
+
+The behavior of field within an API, whether it is required, whether it is only
+presented as output and not accepted as input, or vice versa, is documented with
+the `google.api.field_behavior` annotation as described in [AIP-203][]. Changing
+the value of this annotation can represent a semantic change in the API that is
+percevied as incompatible for existing clients. The following are examples of
+backwards incompatible changes with `google.api.field_behavior`:
+
+* Adding `REQUIRED` to an existing field previously considered `OPTIONAL`
+(implicitly or otherwise).
+* Adding `OUTPUT_ONLY` to an existing field previously accepted as input
+* Removing `OUTPUT_ONLY` from an existing field previously ignored as input
+* Adding `INPUT_ONLY` to an existing field previously emitted as output
+* Adding `IMMUTABLE` to an existing field previously considered mutable
+
+There are some changes that *are* compatible, which are as follows:
+
+* Adding `OPTIONAL` to an existing field
+* Changing from `REQUIRED` to `OPTIONAL` on an existing field
+* Removing `REQUIRED` from an existing field
```

Comment:
wouldn't this be the same as REQUIRED -> OPTIONAL? seems like you don't need to enumerate the "unannotated" case. especially since we require field behavior now.

Reply:
> wouldn't this be the same as REQUIRED -> OPTIONAL?

yes

> seems like you don't need to enumerate the "unannotated" case

I'm choosing to be overly explicit because i've noticed others not make that same distinction.

> especially since we require field behavior now.

Yeah, but there are still many many packages that haven't caught up to this. Doesn't harm to be explicit here imo, especially if there are still cases of it looming. WDYT?

Reply:
yeah fair enough, I suppose being explicit is better.